File: tests/libYARP_run/module.cpp
/*
 * SPDX-FileCopyrightText: 2006-2021 Istituto Italiano di Tecnologia (IIT)
 * SPDX-FileCopyrightText: 2006-2010 RobotCub Consortium
 * SPDX-License-Identifier: BSD-3-Clause
 */

#include <cstdio>
#include <yarp/os/RFModule.h>
#include <yarp/os/Network.h>

using namespace yarp::os;

class MyModule:public RFModule
{
    Port handlerPort; //a port to handle messages
    int count;
public:

    double getPeriod() override
    {
        return 1; //module periodicity (seconds)
    }

    /*
     * This is our main function. Will be called periodically every getPeriod() seconds.
     */
    bool updateModule() override
    {
        count++;
        printf("[%d] updateModule\n", count);

        return true;
    }

    /*
     * Message handler. Just echo all received messages.
     */
    bool respond(const Bottle& command, Bottle& reply) override
    {
        printf("Got something, echo is on\n");
        if (command.get(0).asString()=="quit")
            return false;
        else
            reply=command;
        return true;
    }

    /*
     * Configure function. Receive a previously initialized
     * resource finder object. Use it to configure your module.
     * Open port and attach it to message handler.
     */
    bool configure(yarp::os::ResourceFinder &rf) override
    {
        count=0;
        // handlerPort.open("/regression-test-module");
        // attach(handlerPort);
        // attachTerminal();
        return true;
    }

    /*
     * Interrupt function.
     */
    bool interruptModule() override
    {
        printf("Interrupting your module, for port cleanup\n");
        return true;
    }

    /*
     * Close function, to perform cleanup.
     */
    bool close() override
    {
        printf("Calling close function\n");
        handlerPort.close();
        return true;
    }
};

int main(int argc, char * argv[])
{
    Network yarp;
    MyModule module;

    yarp::os::ResourceFinder rf; //dummy resource finder, empty

    //open a file

    printf("Configure module...\n");
    module.configure(rf);
    printf("Start module...\n");
    module.runModule();

    //remove file
    printf("Main returning...\n");
    return 0;
}

File: src/devices/fake/fakePythonSpeechTranscription/Module.py
#!/usr/bin/env python

# SPDX-FileCopyrightText: 2024-2024 Istituto Italiano di Tecnologia (IIT)
# SPDX-License-Identifier: BSD-3-Clause

def open():
    print("Calling open() function")
    return True

def close():
    print("Calling close() function")
    return True

class SpeecTranscriptor:
    def __init__(self, lang, verbose):
        self.lang = lang
        self.verbose = verbose
        print ('Created a SpeecTranscriptor with language: ' , self.lang, ' verbose: ', self.verbose)

    def set_language(self, new_lang):
        self.lang = new_lang
        print ('Setting Language: ', self.lang)
        return True

    def get_language(self):
        print ('Returning Language: ', self.lang)
        return self.lang

    def set_verbose(self, verbose):
        self.verbose = verbose
        print ('Setting verbose: ', self.verbose)
        return True

    def get_verbose(self):
        print ('Getting verbose: ', self.verbose)
        return self.verbose
